File: applications/dotnet/webdavclientlibrary/Response/Multistatus.cs
/***************************************************************************
 * Copyright 2001-2007 The eXo Platform SAS          All rights reserved.  *
 * Please look at license.txt in info directory for more license detail.   *
 **************************************************************************/
using System;
using System.Collections.Generic;
using System.Text;
using System.Xml;
using System.Collections;

/**
 * Created by The eXo Platform SARL
 * Authors : Vitaly Guly <gavrik-vetal@ukr.net/mail.ru>
 *         : Max Shaposhnik <uy7c@yahoo.com>
 * @version $Id:
 */

namespace exo_jcr.webdav.csclient.Response
{
    public class Multistatus
    {

        protected ArrayList responses = new ArrayList();

        public Multistatus(XmlTextReader reader) {
            while (reader.Read()) {
                switch (reader.NodeType) {
                    case XmlNodeType.Element:
                        if (reader.Name.EndsWith(DavProperty.RESPONSE)) {
                            responses.Add(new DavResponse(reader));
                        }
                        break;
                    case XmlNodeType.EndElement:
                        if (reader.Name.EndsWith(DavProperty.MULTISTATUS)) {
                            return;
                        }
                        throw new XmlException("Malformed response at line " + reader.LineNumber + ":" + reader.LinePosition, null);
                }
            }
        }

        public ArrayList getResponses()
        {
            return responses;
        }

    }
}

File: component/webdav/src/main/java/org/exoplatform/services/webdav/common/response/MultiStatus.java
/**
* Copyright 2001-2006 The eXo Platform SARL         All rights reserved.  *
* Please look at license.txt in info directory for more license detail.   *
*/

package org.exoplatform.services.webdav.common.response;

import java.util.ArrayList;

import org.exoplatform.services.webdav.DavConst;
import org.exoplatform.services.webdav.common.document.AbstractXmlSerializable;
import org.w3c.dom.Document;
import org.w3c.dom.Element;

/**
 * Created by The eXo Platform SARL
 * Author : Vitaly Guly <gavrik-vetal@ukr.net/mail.ru>
 * @version $Id: MultiStatus.java 12899 2007-02-20 15:13:30Z gavrikvetal $
 */

public class MultiStatus extends AbstractXmlSerializable {
  
  private ArrayList<MultiStatusResponse> responses = null;
  
  public MultiStatus(ArrayList<MultiStatusResponse> responses) {
    this.responses = responses;
  }

  public Element createElement(Document document) {
    return document.createElementNS(DavConst.DAV_NAMESPACE, DavConst.DAV_PREFIX + DavConst.DavProperty.MULTISTATUS);
  }
  
  public void serializeBody(Element element) {
    for (MultiStatusResponse curResponse : responses) {
      curResponse.serialize(element);    
    }    
  }
  
}
